terraformCmd.py

```diff
@@ -0,0 +1,111 @@
+#!/usr/bin/env python3
+#
+# Copyright © 2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Based on https://github.com/jlausuch/qatrfm-lib/blob/master/qatrfm/environment.py
+
+""" Terraform environment
+
+Defines how the Terraform deployment looks like and implements the
+appropriate calls to deploy and destroy the environment. All the possible
+parameters are passed to the corresponding .tf file
+
+"""
+
+import json
+import os
+import shutil
+import string
+import sys
+import random
+import tempfile
+import time
+import signal
+import subprocess
+from threading import Timer
+
+from pathlib import Path
+
+
+class TrfmCommandTimeout(Exception):
+    pass
+
+
+class TrfmCommandFailed(Exception):
+    pass
+
+
+class TerraformCmd:
+
+    def __init__(self, logger, tf_file, tf_vars=None, timeout=300):
+        self.logger = logger
+        self.tf_file = tf_file
+        self.timeout = timeout
+        self.logger.info("Terraform TF file: {}".format(self.tf_file))
+        if tf_vars:
+            self.update_tf_vars(tf_vars)
+        self.workdir = tempfile.mkdtemp()
+        self.logger.debug("Using working directory {}".format(self.workdir))
+        shutil.copy(self.tf_file, self.workdir + '/env.tf')
+
+    def update_tf_vars(self, vars):
+        self.tf_vars = ''
+        for v in vars:
+            kv = v.split('=', 1)
+            if (Path(kv[1]).is_file()):
+                kv[1] = Path(kv[1]).resolve()
+            self.tf_vars += "-var '{}={}' ".format(kv[0], kv[1])
+
+    def execute_bash_cmd(self, cmd):
```

Comment:
I would rather keep this outside this class, as it doesn't have anything to do with terraform.

Reply:
yeah sure make sense . for me reasoning behind was that I can omit two input params : `workdir` and `logger` . But maybe it is not good enough reason to insert this here ...

Reply:
another option is to make it static, but still don't like to keep it in this class. But no worries about this, can be changed later on, it's not critical, just a design improvement, so no sweet.